cpp/include/cudf/detail/utilities/device_atomics.cuh

```diff
@@ -116,18 +116,17 @@ struct genericAtomicOperationImpl<T, Op, 4> {
     using T_int = unsigned int;
```

Comment:
We should use a type with a guaranteed width like `uint32_t` instead of a platform-defined width like `unsigned int`.
```suggestion
    using T_int = uint32_t;
```